common/resource/fx.go

```diff
@@ -473,5 +474,12 @@ func getFrontendConnectionDetails(
 
 	}
 
-	return frontendURL, frontendHTTPURL, frontendTLSConfig, nil
+	var frontendHTTPPort int
+	if hasIFE {
+		frontendHTTPPort = cfg.Services[string(primitives.InternalFrontendService)].RPC.HTTPPort
```

Comment:
I don't like this precedent of reaching into other services config sections here. Currently (before this) each service only looked at its own section (and just the presence of the IFE section, but that's a hack). Yes they also look to get a default port if the rpc port is missing in membership but it should always be there so that's never used.

How about we pass the http port through membership?

Also, if cfg.PublicClient.HTTPHostPort is set, it should take the port from that and ignore everything else.